### PR TITLE
ByteVector.toInputStream method

### PR DESCRIPTION
**Pull request by jendakol**
Hello,
I needed to convert the `ByteVector` to `InputStream` and surprisingly, that is not supported yet. Here is the implementation so everyone could use that.

### CODE REVIEW

**Comment by mpilquist** [comment]
Thanks! Looks like the tests are using some methods added in Java 9. Can you adjust so they build on Java 8?

File: core/shared/src/test/scala/scodec/bits/ByteVectorTest.scala [hunk 1]
```diff
@@ -700,4 +700,53 @@ class ByteVectorTest extends BitsSuite {
...
+  test("toInputStream") {
+    {
+      val is = ByteVector(0, 1, 2).toInputStream
+      assert(is.available() == 3)
+      assert(Source.fromInputStream(is).map(_.toInt.toString).mkString == "012")
+    }
+    {
+      val is = ByteVector((0 to 9).toArray.map(_.toByte)).toInputStream
+      assert(is.available() == 10)
+      is.readNBytes(5)
```
Comment (sideeffffect) [inline core/shared/src/test/scala/scodec/bits/ByteVectorTest.scala:713]: `readNBytes` is available only since Java 11

Comment (jendakol) [reply core/shared/src/test/scala/scodec/bits/ByteVectorTest.scala]: Oops.

**Comment by jendakol** [comment]
Should be fixed. Sorry, I didn't realize I have JDK11 as the default one 😄

**Comment by jendakol** [comment]
OK, so another fix attempt, this time for the Scala 2.11 dinosaur 😄

**Comment by mpilquist** [comment]
Looks like Scala.js issue now, perhaps just test related? Not sure, didn't look too closely.

File: core/shared/src/test/scala/scodec/bits/ByteVectorTest.scala [hunk 2]
```diff
@@ -33,7 +33,7 @@ package scodec.bits
 import java.io.ByteArrayOutputStream
 import java.nio.ByteBuffer
 import java.util.{Arrays, UUID}
-
+import scala.io.Source
```
Comment (sideeffffect) [inline core/shared/src/test/scala/scodec/bits/ByteVectorTest.scala:36]: @jendakol this seems to cause the issues with Scala.js
https://stackoverflow.com/a/40167262

But maybe there's a way to write the test that won't require `scala.io.Source`.

**Comment by jendakol** [comment]
Hi @mpilquist, I did some corrections so all tests are green on my machine. I used a small hack to solve one issue with a missing method in `AtomicInteger` in Scala.js - please see the code.

**Comment by jendakol** [comment]
Thank you 🤝

**Pull request merged**